doc/OnlineDocs/library_reference/kernel/index.rst

```diff
@@ -91,6 +91,96 @@ A simplified version of this using :python:`pyomo.environ` components might look
 
 The transformer expressed using :python:`pyomo.kernel` components requires roughly 2 KB of memory, whereas the :python:`pyomo.environ` version requires roughly 8.4 KB of memory (an increase of more than 4x). Additionally, the :python:`pyomo.kernel` transformer is fully compatible with all existing :python:`pyomo.kernel` block containers.
 
+Direct Support For Conic Constraints with Mosek
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Pyomo 5.6.2 introduced support into :python:`pyomo.kernel`
+for six conic constraint forms that are directly recognized
+by the new Mosek solver interface. These are
+
+ - :python:`conic.quadratic`:
+
+     :math:`\;\;\sum_{i}x_i^2 \leq r^2,\;\;r\geq 0`
+
+ - :python:`conic.rotated_quadratic`:
+
+     :math:`\;\;\sum_{i}x_i^2 \leq 2 r_1 r_2,\;\;r_1,r_2\geq 0`
+
+ - :python:`conic.primal_exponential`:
+
+     :math:`\;\;x_1\exp(x_2/x_1) \leq r,\;\;x_1,r\geq 0`
+
+ - :python:`conic.primal_power` (:math:`\alpha` is a constant):
+
+     :math:`\;\;||x||_2 \leq r_1^{\alpha} r_2^{1-\alpha},\;\;r_1,r_2\geq 0,\;0 < \alpha < 1`
+
+ - :python:`conic.dual_exponential`:
+
+     :math:`\;\;-x_2\exp((x_1/x_2)-1) \leq r,\;\;x_2\leq0,\;r\geq 0`
+
+ - :python:`conic.dual_power` (:math:`\alpha` is a constant):
+
+     :math:`\;\;||x||_2 \leq (r_1/\alpha)^{\alpha} (r_2/(1-\alpha))^{1-\alpha},\;\;r_1,r_2\geq 0,\;0 < \alpha < 1`
+
+Other solver interfaces will treat these objects as general
+nonlinear or quadratic constraints, and may or may not have
+the ability to identify their convexity. For instance,
+Gurobi will recognize the expressions produced by the
+:python:`quadratic` and :python:`rotated_quadratic` objects
+as representing convex domains as long as the variables
+involved satisfy the convexity conditions. However, other
+solvers may not include this functionality.
+
+Each of these conic constraint classes are of the same
+category type as standard :python:`pyomo.kernel.constraint`
+object, and, thus, are directly supported by the standard
+constraint containers (:python:`constraint_tuple`,
+:python:`constraint_list`, :python:`constraint_dict`).
+
+Each conic constraint class supports two methods of
+instantiation. The first method is to directly instantiate
+a conic object, providing all necessary input variables:
+
+.. code-block:: python
+
+   import pyomo.kernel as pmo
```

Comment:
Any objection to including these as doctests instead of code-blocks?

Reply:
No objection. I can do that.